web/src/settings_ui.ts

```diff
@@ -18,24 +28,23 @@ export const strings = {
     failure_html: $t_html({defaultMessage: "Save failed"}),
     saving: $t({defaultMessage: "Saving"}),
 };
-
 // Generic function for informing users about changes to the settings
 // UI.  Intended to replace the old system that was built around
 // direct calls to `ui_report`.
 export function do_settings_change(
-    request_method,
-    url,
-    data,
-    status_element,
+    request_method: AjaxRequestHandler,
+    url: string,
+    data: string | Record<string, unknown> | unknown[] | undefined,
```

Comment:
We could also type this parameter like this, since we are directly passing data to the `request_method`.
```suggestion
    data: Parameters<AjaxRequestHandler>[0]["data"],
```

Reply:
Right, I also this in dialog_widget.ts. I was not wonder how would you be sure that `data` will be one of the keys in `Parameters<AjaxRequestHandler>[0]`?

Reply:
We have `data` key in the `args` parameter of `AjaxRequestHandler` that should be enough for ts compiler to infer that `data` will always be a key in `Parameters<AjaxRequestHandler>[0]`

Reply:
fixed that, thanks!